src/omv/sensor.c

```diff
@@ -1192,7 +1284,12 @@ int sensor_snapshot(sensor_t *sensor, image_t *image, streaming_cb_t streaming_c
             }
             break;
         default:
-            return -1;
+            return -1; // Error out if the pixformat is not set.
+    }
+
+    // Error out if the frame size wasn't set or the line width is larger than the camera line buffers.
+    if ((!length) || (((length / h) > (OMV_LINE_BUF_SIZE / 2)) && (addr == ((uint32_t) &_line_buf)))) {
+        return -1;
```

Comment:
Note if length is not set it will return -1 (see line 1287), plus length is not initialized anyway
```suggestion
    // Error out if the line width is larger than the camera line buffer, in line transfer mode
    if (((length / h) > (OMV_LINE_BUF_SIZE / 2)) && (addr == ((uint32_t) &_line_buf))) {
        return -1;
```

Reply:
1287 only bails out if pixformat wasn't set. This is to handle framesize not being set. You have to check if the length is zero and bail on that. Otherwise, you'll try to start a zero-length DMA transfer.